objects/src/accounts/delta/storage.rs

```diff
@@ -154,7 +176,101 @@ impl Deserializable for AccountStorageDelta {
             updated_items.push((idx, value));
         }
 
-        Ok(Self { cleared_items, updated_items })
+        // deserialize and validate storage map deltas
+        let num_updated_maps = source.read_u8()? as usize;
+        let mut updated_maps = Vec::with_capacity(num_updated_maps);
+        for _ in 0..num_updated_maps {
+            let idx = source.read_u8()?;
+            let value = StorageMapDelta::read_from(source)?;
+            updated_maps.push((idx, value));
+        }
+
+        Ok(Self {
+            cleared_items,
+            updated_items,
+            updated_maps,
+        })
+    }
+}
+
+// STORAGE MAP DELTA
+// ================================================================================================
+
+/// [StorageMapDelta] stores the differences between two states of account storage maps.
+///
+/// The differences are represented as follows:
+/// - leave updates: represented by `cleared_leaves` and `updated_leaves` field.
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+pub struct StorageMapDelta {
+    pub cleared_leaves: Vec<Word>,
+    pub updated_leaves: Vec<(Word, Word)>,
+}
+
+impl StorageMapDelta {
+    /// Creates a new storage map delta from the provided cleared and updated leaves.
+    pub fn from(cleared_leaves: Vec<Word>, updated_leaves: Vec<(Word, Word)>) -> Self {
+        StorageMapDelta {
+            cleared_leaves: Vec::from(cleared_leaves),
+            updated_leaves: Vec::from(updated_leaves),
+        }
+    }
+
+    /// Checks whether this storage map delta is valid.
+    ///
+    /// # Errors
+    /// Returns an error if:
+    /// - Any of the cleared or updated leaves is referenced more than once (e.g., updated twice).
+    pub fn validate(&self) -> Result<(), AccountDeltaError> {
```

Comment:
@bobbinth @hackaugusto here it is not super clear to me what we want to validate. I summarized the question here https://github.com/0xPolygonMiden/miden-base/issues/382#issuecomment-2006625375